oops/backplane/gold_master.py

```diff
@@ -0,0 +1,2112 @@
+                    comparison.status = 'Mask mismatch'
+                else:
+                    comparison.status = 'Success'
+
+        # Determine status
+        success = comparison.success
+        level = logging.INFO if success else logging.ERROR
+
+        # Construct the message
+        message = [comparison.suite, ' | ', comparison.status]
+
+        # Handle failed comparison cases
+        if errors2 == 0 and comparison.status != 'Success':
+            if comparison.pickle_path:
+                basename = os.path.basename(comparison.pickle_path)
+                message += [': ', basename]
+
+            else:
+                message += [': "', comparison.title, '"']
+
+        # Handle comparison cases
+        else:
+            val = self._valstr(comparison.max_diff1)
+            message += [': "', comparison.title, '"; diff=',
+                        self._valstr(comparison.max_diff1)]
+
+            if comparison.max_diff2 != comparison.max_diff1:
+                if comparison.max_diff2 <= comparison.limit * 1.e-8:
+                    message += ['/0.']
+                else:
+                    message += ['/', self._valstr(comparison.max_diff2)]
+
+            message += ['/', self._valstr(comparison.limit)]
+
+            if comparison.distance:
+                message += ['; offset=', self._valstr(comparison.distance),
+                            '/', self._valstr(comparison.radius)]
+
+            message += ['; pixels=', str(errors1)]
+
+            if errors2 != errors1:
+                message += ['/', str(errors2)]
+
+            message += ['/', str(comparison.pixels)]
+
+        LOGGING.print(''.join(message), level=level)
+
+        # Save the results
+        self.results[comparison.title] = comparison
+
+    #===========================================================================
+    @staticmethod
+    def _valstr(value):
+        """value formatter, avoiding "0.000" and "1.000e-12"."""
+
+        if isinstance(value, Qube):
+            if value.mask:
+                return '--'
+            value = value.vals
+
+        if isinstance(value, numbers.Integral):
+            return str(value)
+        else:
+            formatted = '%#.4g' % value
+            parts = formatted.partition('e')
+            return parts[0].rstrip('0') + parts[1] + parts[2]
+
+    #===========================================================================
+    @functools.lru_cache(maxsize=10)
+    def _footprint(radius):
+        """Circular footprint of the given pixel radius, for scip.ndarray
+        filters.
+        """
+
+        rounded = int(radius // 1)      # rounded down
+        size = 2 * rounded + 1
+        x = np.arange(size) - size//2   # [-N to N] inclusive
+        y = x[:, np.newaxis]
+        return (x**2 + y**2) <= radius**2
+
+    #===========================================================================
+    def _basename(self, title, gold=False):
```

Comment:
The new Linux-friendly names are not terrible, I guess, so I suppose I could go along. However, I really do prefer the more readable names on my Mac. This seemed like a way to thread the needle. I guess the issue for you would be that you run the tests on Windows but don't want the Windows versions of the names, and also don't want to have to type "--platform linux". If that's so, we can change what it does by default on Windows because I really don't care. Let me know.

Reply:
I don't think anybody is currently running this on Windows. It just seemed like an awful lot of effort. There's no other software we run that does this kind of change between platforms. I'm also thinking about use by other people in the future, since this is supposed to be open-source software and we hope that it will be used by many people. Having the filenames change based on the current platform just seems confusing. Would you be OK with making the default by Linux all the time, and then you can specify a command line option to make it MacOS for your own use, since you're probably the only person who will ever want that?

Reply:
I think where we are now is a decent compromise between your needs and mine. I haven't had any issues at all with one set of more readable filenames in my local test directory and another in the master directory. Management of the master copies is invisible to the user. If some future Mac or Windows user has a different preference, we can always revisit the issue then.